contributing.md

```diff
@@ -0,0 +1,146 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change. 
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Filing an issue
+
+1. Look for a similar issue that may already describe the proposed changes.
+
+1. If none exists, create a new issue and describe the proposed changes.
+
+1. Apply any relevant labels to the issue and set the milestone to Backlog.
+
+## Pull Request Process
+
+1. Find the issue that describes the changes you are going to make.  If no issue exists
+for the changes you want to make, see above and file an issue describing the changes, then
+continue to the next step.
+
+1. Assign the issue to yourself, and change the milestone from Backlog to the currently active
+milestone.
+
+1. Create a branch in your local fork with the issue number in the name.
+
+1. Implement the code changes described by the issue.  All Python code changes should
+be accompanied by any relevant tests, either by modifying existing ones or adding new 
+ones.  Front-end changes (HTML/CSS/JS), and infrastructure changes (Docker related things)
+do not require tests.
+
+1. When all your changes are complete to your satisfaction, run `make check` to run the
+full test, linting, formatting, and coverage suites.  Fix any linting issues or breaking tests, 
+and make sure coverage remains at 100% by adding new tests for any uncovered lines.
+
+1. When all tests and checks are passing, commit all your changes into a single commit and name 
+the commit with a brief description of the change and 'fixes #123' where #123 is the issue number.
+You can add any additional descriptive information to the message body of the commit, but the issue 
+number must be present in the first line of the commit.  Example: `Add new dashboard link fixes #123`
+
+1. Push your branch up to your fork and submit a pull request on to master.  Add any additional 
+information you'd like to the pull request body, including descriptions of changes, screenshots
+of any UI changes, special instructions for testing, etc. 
+
+1. Find a reviewer.  If you're not sure who should review, please contact us on #experimenter on
+[IRC](https://wiki.mozilla.org/IRC).
+
+1. If you receive feedback that requires changes to your pr, make the changes locally, and then 
+ammend them to the commit you submitted rather than adding additional commits.  A PR will only be
```

Comment:
I personally don't like this at all. I'm ok with squashing the PRs at the end, but squashing during review puts a lot of burden on the reviewer to untangle the differences between revisions. Github is getting better at this, but it's still not good enough to demand every revision be squashed during review in my opinion.

Reply:
@mythmon Good point, if we can accomplish the same effect by using the 'Squash and Rebase' button then squashing before pushing isn't necessary.